src/cmd-buildextend-metal

```diff
@@ -143,6 +143,17 @@ if [ "${image_type}" = dasd ] || [ "${image_type}" = metal4k ]; then
     ignition_platform_id=metal
 fi
 
+# Firstboot kernel arguments.
+firstboot_kargs="$(python3 -c '
+import sys, yaml;
+LEGACY_DEFAULT = [ "rd.neednet=1", "ip=dhcp,dhcp6" ];
+args = yaml.safe_load(sys.stdin).get("extra-kargs", LEGACY_DEFAULT);
```

Comment:
Hmm, I don't follow this. `extra-kargs` are persistent, but we want `ip=dhcp,dhcp6` to just be on first boot, right?

I think it might be safer/cleaner to just have a `rd-networking-by-default: true/false` type knob instead.

Reply:
That's actually a typo (I was planning to finish and test this PR tomorrow), it should be `firstboot-kargs`. It couples with a `image.yaml` snippet like this:
```yaml
firstboot-kargs:
    - 'rd.neednet=1'
    - 'ip=dhcp,dhcp6'
```

(None of this code has been manually tested, yet)

Reply:
Ahh OK, that makes more sense. In that case, WFM. (We'll want to make sure though that we add this to the YAML files in both FCOS and RHCOS before merging.)

Reply:
Hmm, thinking more on this, this still feels a bit awkward to me. Can we instead drop the `LEGACY_DEFAULT` here and just add the snippet in your comment above in `image.yaml` before merging this? Otherwise, the "pivot" point in FCOS/RHCOS would be setting `firstboot-kargs: []`, right? I think it's cleaner and clearer for the pivot to be when we remove `firstboot-kargs` entirely.

Reply:
Eh, we can, but this PR still has a dependency on RHCOS config plus all versioned streams.

I'd suggest we keep the `LEGACY_DEFAULT` as is for now and merge this first, then propagate the image.yaml change to all the relevant repos to make sure the defaults are in place, then possibly came back here and address the pivot point (I'm fine to handle the "no-key" case as the "empty value" case, but we should maybe leave a commented key-sample once we get there).

Reply:
OK sure, WFM. I think this is the right approach overall (make default configurable, then configure it in the streams that want in, then eventually flip default here so we can drop the config knob) and we're just bikeshedding on the type of the knob. :)

See for example https://github.com/coreos/fedora-coreos-config/pull/426/files#diff-b56ad46b43eb108be045b7ad559c32b6R5. I think it would've been more obvious if the diff there were e.g.

```diff
+rd-networking-by-default: false
```

But anyway, we should be able to blow away that knob entirely soon-ish!